src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/ProtobufOtlpGrpcExportClient.cs

```diff
@@ -44,25 +52,115 @@ public ExportClientResponse SendExportRequest(byte[] buffer, int contentLength,
         try
         {
             using var httpRequest = this.CreateHttpRequest(buffer, contentLength);
-
             using var httpResponse = this.SendHttpRequest(httpRequest, cancellationToken);
 
             try
             {
                 httpResponse.EnsureSuccessStatusCode();
             }
-            catch (HttpRequestException ex)
+            catch (HttpRequestException)
+            {
+                throw;
+            }
+
+            var trailingHeaders = httpResponse.TrailingHeaders();
+            Status? status = GrpcProtocolHelpers.GetResponseStatus(httpResponse, trailingHeaders);
+
+            if (status == null)
+            {
+                using var responseStream = httpResponse.Content.ReadAsStreamAsync().GetAwaiter().GetResult();
+                int firstByte = responseStream.ReadByte();
+
+                if (firstByte == -1)
+                {
+                    if (status?.StatusCode == StatusCode.OK)
+                    {
+                        status = new Status(StatusCode.Internal, "Failed to deserialize response message.");
+                    }
+
+                    OpenTelemetryProtocolExporterEventSource.Log.ResponseDeserializationFailed(this.Endpoint.ToString());
+
+                    return new ExportClientGrpcResponse(
+                        success: false,
+                        deadlineUtc: deadlineUtc,
+                        exception: null,
+                        status: status,
+                        grpcStatusDetailsHeader: null);
+                }
+
+                trailingHeaders = httpResponse.TrailingHeaders();
+                status = GrpcProtocolHelpers.GetResponseStatus(httpResponse, trailingHeaders);
```

Comment:
We did this work on line 66 do we need to do it again?

Reply:
The second call is necessary because trailing headers might not be fully available until the response stream is consumed. While we retrieve the status initially, gRPC often sends critical information like error details or final statuses in trailing headers, which can only be reliably accessed after reading the response body. This additional check ensures we handle cases where the gRPC status or errors are deferred until the stream is processed, improving robustness and accuracy. This follows the same logic as in Grpc library - https://github.com/grpc/grpc-dotnet/blob/5a58c24efc1d0b7c5ff88e7b0582ea891b90b17f/src/Grpc.Net.Client/Internal/GrpcCall.cs#L465

Reply:
👍 I pushed a commit to capture that info as a comment